include/ttmlir/Dialect/TTIR/IR/TTIRBase.td

```diff
@@ -41,6 +43,28 @@ def TTIR_Dialect : Dialect {
 //===----------------------------------------------------------------------===//
 
 class TTIR_Op<string mnemonic, list<Trait> traits = []> :
-        Op<TTIR_Dialect, mnemonic, !listconcat(traits, [Pure])>;
+        Op<TTIR_Dialect, mnemonic, !listconcat([Pure], traits)>;
```

Comment:
I've changed the order of `listconcat`arguments everywhere, because of dependent trait list for traits like `TTIR_Involution`. Otherwise it fails long before compiling (it check for existence of this traits while still parsing `.td` file), so it doesn't see a trait/interface like `DestinationStyleOpInterface` on ops that has it, because it builds list in bottom-up fashion (`[trait_of_op, trait_of_ops_class, trait_op_ops_class_class,...]`).

From what I saw it doesn't make any difference in generated C++ code. @nsmithtt Did you have an experience with this?